osf/metrics/reporters/spam_count.py

```diff
@@ -0,0 +1,76 @@
+from osf.models import OSFUser
+
+from osf.metrics.reports import SpamReport
+from ._base import MonthlyReporter
+from osf.models import PreprintLog, NodeLog
+from osf.models.spam import SpamStatus
+
+import dateutil
+from django.utils import timezone
+
+
+class SpamCountReporter(MonthlyReporter):
+
+    def report(self, report_date):
+        today = timezone.now()
```

Comment:
`report_date` will be a `YearMonth` for last month. I'd extend `YearMonth` slightly, as follows

```
class YearMonth(typing.NamedTuple):
    year: int
    month: int

    def __str__(self):
        return f'{self.year}-{self.month}'

    def target_month(self):
        return datetime.datetime(self.year, self.month, 1, tzinfo=pytz.utc)

    def next_month(self):
        if self.month == 12:
            return datetime.datetime(self.year + 1, 1, 1, tzinfo=pytz.utc)
        return datetime.datetime(self.year, self.month + 1, 1, tzinfo=pytz.utc)
```

Allowing this to be 


```
def report(self, report_yearmonth: YearMonth):
    target_month = report_yearmonth.target_month()
    next_month = report_yearmonth.next_month()
    #[...]
```